### DISCUSSION TITLE
Chapter 13: Converting the App to a Progressive Web App

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#136] [Going the Distance Discussion]
Topics:

- Intro to PWA's
- Converting Space-Truckers to a PWA
- Using IndexedDB to store High Scores
- Storing and Displaying High Scores

**Comment by m-size** [comment]
is the service-worker.js automatically generated or not needed at all?
I cannot find the worker-service.js on the repository.

**Reply by jelster** [reply]
Hi @kusamochi - thanks for posting!
As you suspected, it is because the list of URL's and assets is dynamically discovered and injected into the service-worker.js file that the file does not appear in the repository and instead is generated by the WorkBox WebPack plugin. This plugin is configured in the `./webpack.common.js` module, which deposits the generated ServiceWorker into the `/dist` folder as part of the overall packaging process. 

Though the `service-worker.js` and the associated `workbox-<hash>.js` files are minified, the Service Worker definition lists the URL's and assets to cache in a call like this:

```javascript
   e.precacheAndRoute(
      [
        {
          url: "assets/2k_ceres_fictional.jpg",
          revision: "f8bb7ef71b1e29a6cabf59dfb6f9e38f",
        },
        {
          url: "assets/2k_sun.jpg",
          revision: "7ce3a845bb3ca672ddd6660fd477de1f",
        },
// ...
```



If you want to view the generated file, you can invoke the `npm run build` command from the root of the source repository. 

Hope this answers your question, and thanks for contributing to the community!

**Reply by m-size** [reply]
oh, thank you very much for your response. I will look into it :)
